### PR TITLE
Document the connection path for podman --remote

### PR DESCRIPTION
**Pull request by rhatdan**
Issue: GH-7010
Signed-off-by: Jhon Honce <jhonce@redhat.com>
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Replaces: https://github.com/containers/podman/pull/7118

**Comment by rhatdan** [comment]
@jwhonce @mheon @vrothberg @giuseppe @ashley-cui @baude @TomSweeneyRedHat @QiWang19 PTAL

**Comment by rhatdan** [comment]
I am not sure we want to support DOCKER_HOST at all.  

The issue with the original PR is that if you have podman and docker on the same remote client, you could have Podman attempting to talk to the Docker daemon IF DOCKER_HOST was defined even if you had a valid connection in containers.conf.

I think we ignore this environment variable if set.

I cleaned up the documentation and added it to the podman-remote man page.

**Comment by rhatdan** [comment]
@afbjorklund PTAL

**Comment by rhatdan** [comment]
@AkihiroSuda  PTAL

**Review by AkihiroSuda — APPROVED** [review]

File: docs/source/markdown/podman-remote.1.md [hunk 1]
```diff
@@ -39,6 +39,12 @@ Path to ssh identity file. If the identity file has been encrypted, Podman promp
 If no identity file is provided and no user is given, Podman defaults to the user running the podman command.
 Podman prompts for the login password on the remote server.
 
+Identity value resolution precedence:
+ - command line value
+ - environment variable `CONTAINER_SSHKEY`, if `CONTAINER_HOST` is found
+ - `containers.conf`
+ - `""`
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-remote.1.md:46]: IDK, but maybe this
```suggestion
 - `""` (default value)
```

and if so, similar change for the URL below.

**Comment by TomSweeneyRedHat** [comment]
And what's wrong with the Podman CI today?  Almost every PR has happy green test buttons?  ;^)

File: docs/source/markdown/podman-remote.1.md (hunk 1 above)
Comment (rhatdan) [reply docs/source/markdown/podman-remote.1.md]: I am just going to remove them, they don't really do anything.  I guess Podman remote will prompt if the user specifies nothing.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**